### DISCUSSION TITLE
Message-passing between sibling services

### DISCUSSION THREAD

**Discussion by rajansaini691** [statelyai/xstate#1571] [Q&A]
Hello! Suppose you have a service S, with two invoked child services A, B. If, say, A wanted to send a message to B, is there a way for it to do so directly, or does it have to relay it up to S first with sendParent?

Also, I was wondering if there was a combination of sendParent and forwardTo, like a forwardToParent?

**Comment by Andarist** [comment]
> Hello! Suppose you have a service S, with two invoked child services A, B. If, say, A wanted to send a message to B, is there a way for it to do so directly, or does it have to relay it up to S first with sendParent?

You can share the address of a child within an event sent to its sibling. This would allow for direct sibling-to-sibling communication. An address is always available as `_event.origin` so if child A sends something to its parent then you can store that address that is available on the sent event and share it with child B.

> Also, I was wondering if there was a combination of sendParent and forwardTo, like a forwardToParent?

```js 
forwardTo('#_parent')
```

**Reply by rajansaini691** [reply]
I just wanted to quickly follow up on this one. How would one grab the address from the scxml event in typescript? The typing system seems to strictly enforce two parameters:
```
actions: {
    (ctx: any, evt: any, {_evt}) => { console.log(_evt.origin) },
    assign: { addr: (ctx: any, evt: any, {_evt}) => _evt.origin }
}
```
Both methods refuse to compile. What should be done instead?

**Reply by davidkpiano** [reply]
It's `{ _event }`, not `{ _evt }`.
